### PR TITLE
update soft entities and fix deadlock from qml.

### PR DESCRIPTION
**Pull request by danteruiz**
This PR respects the local translation and orientation of soft entities. The PR also updates disables the is soft option for entity that do not have any joints names of the avatar. This PR also fixes two deadlocks that I experience in updating the avatar qml app.. The deadlock happened in EntitiesScriptingInterface getJointNames and getJointIndex.

tickets 
- https://highfidelity.fogbugz.com/f/cases/17679/Avatarapp-soften-unsoften-wearable-results-in-shifting-disappearing-wearable
https://highfidelity.fogbugz.com/f/cases/17114/soft-entities-can-t-be-edited-when-worn

### CODE REVIEW

**Comment by danteruiz** [comment]
Test plan
- open the avatar app
- click on the avatar entities adjustment bottom (you should not deadlock when pressing this button)
- add this avatar entity to your avatar `https://hifi-content.s3.amazonaws.com/jimi/avatar/photo-real/Clothes/TShirt.fbx`
- select soft option at the bottom. If you update the position and rotation the soft entity should update and rotate.

2. 
- switch to a different avatar
- click on the avatar entities adjustment bottom
- make sure that the avatar has a avatar entitiy
- the soft button option should be disable when the model does not have any joints names that match the avatar joint names. (anime boy or girl has a glasses model avatar entity that is good to test with.
- If you add the shirt entity, the soft button should be enabled.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13411/HighFidelity-Beta-PR14100-c1c8537.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13411/HighFidelity-Beta-PR14100-c1c8537.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13411/HighFidelity-Beta-Interface-PR14100-c1c8537.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13411/HighFidelity-Beta-PR14100-c1c8537.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13411/HighFidelity-Beta-Interface-PR14100-c1c8537.exe

File: interface/resources/qml/hifi/avatarapp/AdjustWearables.qml [hunk 1]
```diff
@@ -100,6 +100,21 @@ Rectangle {
         wearablesModel.setProperty(wearableIndex, 'properties', wearableModelItemProperties);
     }
 
+    function entityHasAvatarJoints(entityID) {
+        console.log(entityID);
```
Comment (sethalves) [inline interface/resources/qml/hifi/avatarapp/AdjustWearables.qml]: remove the log line?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13503/HighFidelity-Beta-PR14100-ad92b17.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13503/HighFidelity-Beta-PR14100-ad92b17.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13503/HighFidelity-Beta-Interface-PR14100-ad92b17.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13503/HighFidelity-Beta-PR14100-ad92b17.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13503/HighFidelity-Beta-Interface-PR14100-ad92b17.exe

**Comment by JeffClinton** [comment]
Currently testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13568/HighFidelity-Beta-PR14100-182c92b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13568/HighFidelity-Beta-PR14100-182c92b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13568/HighFidelity-Beta-Interface-PR14100-182c92b.dmg

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13568/HighFidelity-Beta-Interface-PR14100-182c92b.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13615/HighFidelity-Beta-PR14100-5734c24.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13615/HighFidelity-Beta-PR14100-5734c24.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13615/HighFidelity-Beta-Interface-PR14100-5734c24.exe

**Comment by JeffClinton** [comment]
Currently testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13623/HighFidelity-Beta-PR14100-676293c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13623/HighFidelity-Beta-PR14100-676293c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13623/HighFidelity-Beta-Interface-PR14100-676293c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13623/HighFidelity-Beta-PR14100-676293c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13623/HighFidelity-Beta-Interface-PR14100-676293c.exe

**Comment by JeffClinton** [comment]
Currently testing

File: interface/resources/qml/hifi/avatarapp/AdjustWearables.qml [hunk 2]
```diff
@@ -23,6 +23,9 @@ Rectangle {
     property bool modified: false;
     Component.onCompleted: {
         modified = false;
+        MyAavatar.animGraphLoaded.connect(function() {
```
Comment (wayne-chen) [inline interface/resources/qml/hifi/avatarapp/AdjustWearables.qml]: Typo - should be `MyAvatar`

**Comment by divyaelango88** [comment]
@danteruiz @wayne-chen does it require re-qa?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13655/HighFidelity-Beta-PR14100-613a6ab.apk

**Comment by JeffClinton** [comment]
Currently testing (once built)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13655/HighFidelity-Beta-PR14100-613a6ab.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13655/HighFidelity-Beta-Interface-PR14100-613a6ab.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13655/HighFidelity-Beta-PR14100-613a6ab.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13655/HighFidelity-Beta-Interface-PR14100-613a6ab.exe

**Pull request merged**